src/dlgConnectionProfiles.cpp

```diff
@@ -1662,6 +1676,156 @@ void dlgConnectionProfiles::fillout_form()
+        const QString& profileName = mProfileList.at(i);
+        if (profileName.isEmpty()) {
+            continue;
+        }
+
+        if (hasCustomIcon(profileName)) {
+            loadProfileIcon(font, profileName);
+        } else {
+            // mProfileList is derived from a filesystem directory, but MacOS is not
+            // necessarily case preserving for file names so any tests on them
+            // should be case insensitive
+            // skip creating icons for default MUDs as they are already created above
+            if (mDefaultGames.contains(profileName, Qt::CaseInsensitive)) {
+                continue;
+            }
+
+            generateProfileIcon(font, i, profileName);
+        }
+    }
+}
+
+bool dlgConnectionProfiles::hasCustomIcon(const QString& profileName) const
+{
+    return QFileInfo::exists(mudlet::getMudletPath(mudlet::profileDataItemPath, profileName, QStringLiteral("profileicon")));
+}
+
+void dlgConnectionProfiles::loadProfileIcon(const QFont& font, const QString& profileName) const
+{
+    auto pItem = new QListWidgetItem(profileName);
+    pItem->setFont(font);
+    pItem->setForeground(QColor(Qt::white));
+    auto profileIconPath = mudlet::getMudletPath(mudlet::profileDataItemPath, profileName, QStringLiteral("profileicon"));
+    auto icon = QIcon(QPixmap(profileIconPath).scaled(QSize(120, 30), Qt::IgnoreAspectRatio, Qt::SmoothTransformation).copy());
+    pItem->setIcon(icon);
+    auto description = getDescription(profileName, 0, profileName);
+    if (!description.isEmpty()) {
+        pItem->setToolTip(QLatin1String("<html><head/><body><p>") % description % QLatin1String("</p></body></html>"));
+    }
+    profiles_tree_widget->addItem(pItem);
+}
+
+void dlgConnectionProfiles::generateProfileIcon(const QFont& font, int i, const QString& profileName) const
+{
+    auto pItem = new QListWidgetItem(profileName);
+
+    pItem->setFont(font);
+    pItem->setForeground(QColor(Qt::white));
+    profiles_tree_widget->addItem(pItem);
+    QPixmap background(120, 30);
+    background.fill(Qt::transparent);
+    uint hash = qHash(profileName);
+    QLinearGradient shade(0, 0, 120, 30);
+    quint8 i1 = hash % 255;
+    quint8 i2 = (hash + i) % 255;
+    quint8 i3 = (i * hash) % 255;
+    quint8 i4 = (3 * hash) % 255;
+    quint8 i5 = (hash) % 255;
+    quint8 i6 = (hash / (i + 2)) % 255; // In the other place where this is used i might be -1 or 0
+    shade.setColorAt(1, QColor(i1, i2, i3, 255));
+    shade.setColorAt(0, QColor(i4, i5, i6, 255));
+    QPainter pt(&background);
+    pt.setCompositionMode(QPainter::CompositionMode_SourceOver);
+    pt.fillRect(QRect(0, 0, 120, 30), shade);
+    QPixmap pg(QStringLiteral(":/icons/mudlet_main_32px.png"));
+    pt.drawPixmap(QRect(5, 5, 20, 20), pg);
+
+    QFont _font;
+    QImage _pm(90, 30, QImage::Format_ARGB32_Premultiplied);
+    QPainter _pt(&_pm);
+    _pt.setCompositionMode(QPainter::CompositionMode_SourceOver);
+    int fs = 30;
```

Comment:
why are we initializing the variable outside the loop?

Reply:
:man_shrugging: fixed.